coral/src/app/features/approvals/acls/AclApprovals.tsx

```diff
@@ -33,166 +43,220 @@ interface AclRequestTableData {
-    status: ({ topictype }) => ({
-      status: topictype === "Consumer" ? "success" : "info",
-      text: topictype,
-    }),
-  },
-  {
-    type: "text",
-    field: "username",
-    headerName: "Requested by",
-  },
-  {
-    type: "text",
-    field: "requesttimestring",
-    headerName: "Date requested",
-  },
-  {
-    // Not having a headerName triggers React error:
-    // Warning: Encountered two children with the same key, ``.
-    headerName: "",
-    type: "custom",
-    UNSAFE_render: () => {
-      return (
-        <GhostButton
-          icon={infoSign}
-          onClick={() => alert("Details modal with approve and reject buttons")}
-          title={"View request details"}
-          dense
-        >
-          View details
-        </GhostButton>
-      );
-    },
-  },
-  {
-    width: 30,
-    // Not having a headerName triggers React error:
-    // Warning: Encountered two children with the same key, ``.
-    headerName: "",
-    type: "custom",
-    UNSAFE_render: () => {
-      return (
-        <GhostButton
-          onClick={() => alert("Approve request right away")}
-          title={"Approve request"}
-        >
-          <Icon color="grey-70" icon={tickCircle} />
-        </GhostButton>
-      );
-    },
-  },
-  {
-    width: 30,
-    // Not having a headerName triggers React error:
-    // Warning: Encountered two children with the same key, ``.
-    headerName: "",
-    type: "custom",
-    UNSAFE_render: () => {
-      return (
-        <GhostButton
-          onClick={() => alert("Reject modal with form for reason")}
-          title={"Reject request"}
-        >
-          <Icon color="grey-70" icon={deleteIcon} />
-        </GhostButton>
-      );
-    },
-  },
-];
-
 function AclApprovals() {
+  const queryClient = useQueryClient();
   const [searchParams, setSearchParams] = useSearchParams();
   const initialPage = Number(searchParams.get("page"));
-
   const [activePage, setActivePage] = useState(initialPage || 1);
 
+  const [detailsModal, setDetailsModal] = useState({
+    isOpen: false,
+    reqNo: "",
+  });
+  const [rejectModal, setRejectModal] = useState({ isOpen: false, reqNo: "" });
+
+  const [errorMessage, setErrorMessage] = useState("");
+
   const { data, isLoading } = useQuery<AclRequestsForApprover, Error>({
     queryKey: ["aclRequests", activePage],
     queryFn: () => getAclRequestsForApprover({ pageNo: String(activePage) }),
     keepPreviousData: true,
   });
 
+  const { isLoading: approveIsLoading, mutate: approveRequest } = useMutation({
+    mutationFn: approveAclRequest,
+    onSuccess: (data) => {
+      if (data.result !== "success") {
+        return setErrorMessage(
+          data.message || data.result || "Unexpected error"
+        );
+      }
+      setDetailsModal({ isOpen: false, reqNo: "" });
+      // We need to invalidate the query populating the table to reflect the change
+      queryClient.invalidateQueries(["aclRequests", activePage]);
```

Comment:
I thought we may be able to use set the data for the the query instead here
(https://tanstack.com/query/v4/docs/react/guides/updates-from-mutation-responses) but realised it does not really make sense because our response is paginated. 

But maybe it would be a good idea to at some point think a bit more in detail about how we handle all our queries (refetching and staleTime etc.) 🤔

Reply:
Oooh nice, this seems much better indeed! Will look into it

Reply:
And yes, for sure there is a lot of optimisation potential with that library :o

Reply:
If it works with our paginated responses 🤔  I'm not 1000% sure about that. 
But if it would work it could be noice!

Reply:
Unfortunately it won't work as shown in the resource your linked, as the approve/decline mutations do not return any data, only a standard API response with success or failure. So we can refresh the query data in that spot.

But it seems like it only invalidates the current page of result, the cache for the other pages work fine (no refetch when navigating previously visited pages). So it's ok, I think :o

Reply:
Yes, that sounds good!